source/common/http/utility.cc

```diff
@@ -281,37 +282,65 @@ Utility::parseHttp1Settings(const envoy::api::v2::core::Http1ProtocolOptions& co
   return ret;
 }
 
-void Utility::sendLocalReply(bool is_grpc, StreamDecoderFilterCallbacks& callbacks,
-                             const bool& is_reset, Code response_code, absl::string_view body_text,
-                             const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
-                             bool is_head_request) {
-  sendLocalReply(is_grpc,
+Utility::LocalReplyInfo Utility::generateLocalReplyInfo(const Http::HeaderMap& request_headers) {
+  bool is_grpc = false, is_head_request = false, has_json_content_type = false;
+  // TODO(zyfjeff): This code is copied from Envoy::Grpc::Common::hasGrpcContentType in order to
+  // rely directly on it to avoid causing circular dependencies.
+  const Http::HeaderEntry* content_type = request_headers.ContentType();
+  // Content type is gRPC if it is exactly "application/grpc" or starts with
+  // "application/grpc+". Specifically, something like application/grpc-web is not gRPC.
+  is_grpc =
+      content_type != nullptr &&
+      absl::StartsWith(content_type->value().getStringView(),
+                       Http::Headers::get().ContentTypeValues.Grpc) &&
+      (content_type->value().size() == Http::Headers::get().ContentTypeValues.Grpc.size() ||
+       content_type->value().c_str()[Http::Headers::get().ContentTypeValues.Grpc.size()] == '+');
+
+  if (Http::Headers::get().MethodValues.Head == request_headers.Method()->value().c_str()) {
+    is_head_request = true;
+  }
+
+  if (!is_grpc) {
+    has_json_content_type =
+        content_type != nullptr &&
+        (Http::Headers::get().ContentTypeValues.Json == content_type->value().c_str());
```

Comment:
Instead of looking at the content type of the request, shouldn't we be looking at the `Accept` header here? Also, do we think this overall feature should be configurable? I could go either way, but it definitely needs a release note and needs to be documented.

Reply:
Sorry, I got it wrong, it should be using `Accept` header here.

I don't think there needs to be configured, here just doing a standard HTTP server should do.